packages/headless/src/controllers/commerce/recent-queries-list/headless-recent-queries-list.ts

```diff
@@ -0,0 +1,192 @@
+import {ArrayValue, BooleanValue, NumberValue, Schema} from '@coveo/bueno';
+import {CommerceEngine} from '../../../app/commerce-engine/commerce-engine';
+import {UpdateQueryActionCreatorPayload} from '../../../features/commerce/query/query-actions';
+import {logClearRecentQueries} from '../../../features/commerce/recent-queries/recent-queries-analytics-actions';
+import {recentQueriesReducer as recentQueries} from '../../../features/commerce/recent-queries/recent-queries-slice';
+import {
+  PrepareForSearchWithQueryOptions,
+  executeSearch,
+  prepareForSearchWithQuery,
+} from '../../../features/commerce/search/search-actions';
+import {commerceSearchReducer as search} from '../../../features/commerce/search/search-slice';
+import {
+  clearRecentQueries,
+  registerRecentQueries,
+} from '../../../features/recent-queries/recent-queries-actions';
+import {RecentQueriesSection} from '../../../state/state-sections';
+import {loadReducerError} from '../../../utils/errors';
+import {
+  validateInitialState,
+  validateOptions,
+} from '../../../utils/validate-payload';
+import {
+  buildController,
+  Controller,
+} from '../../controller/headless-controller';
+import type {
+  RecentQueriesListInitialState,
+  RecentQueriesListOptions,
+  RecentQueriesListProps,
+} from '../../recent-queries-list/headless-recent-queries-list';
+
+const defaultRecentQueriesState: Required<RecentQueriesListInitialState> = {
+  queries: [],
+};
+
+const defaultRecentQueriesOptions: Required<RecentQueriesListOptions> = {
+  maxLength: 10,
+  clearFilters: true,
+};
+
+const initialStateSchema = new Schema<RecentQueriesListInitialState>({
+  queries: new ArrayValue({required: true}),
+});
+
+const optionsSchema = new Schema<RecentQueriesListOptions>({
+  maxLength: new NumberValue({required: true, min: 1}),
+  clearFilters: new BooleanValue(),
+});
+
+/**
+ * The `RecentQueriesList` controller manages the user's recent queries.
+ */
+export interface RecentQueriesList extends Controller {
+  /**
+   * The state of the RecentQueriesList controller.
+   * */
+  state: RecentQueriesState;
+  /**
+   * Clears the recent queries list.
+   */
+  clear(): void;
+  /**
+   * Executes the given recent query.
+   * @param index - The index of the recent query to execute.
+   */
+  executeRecentQuery(index: number): void;
+}
+
+/**
+ * A scoped and simplified part of the headless state that is relevant to the `RecentQueriesList` controller.
+ * */
+export interface RecentQueriesState {
+  /**
+   * The list of recent queries.
+   */
+  queries: string[];
+  /**
+   * The maximum number of queries to retain in the list.
+   */
+  maxLength: number;
+  /**
+   * Whether analytics & tracking are enabled.
+   * In the case where it is disabled, it is recommended not to save recent queries.
+   */
+  analyticsEnabled: boolean;
```

Comment:
Why do we need to expose this property on _this_ specific controller? Is it because it's often used in the context of deciding whether we should persist recent queries?